tui/network_dialog_check.go

```diff
@@ -0,0 +1,216 @@
+// Copyright © 2018 Intel Corporation
+//
+// SPDX-License-Identifier: GPL-3.0-only
+
+package tui
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/VladimirMarkelov/clui"
+	term "github.com/nsf/termbox-go"
+
+	"github.com/clearlinux/clr-installer/controller"
+	"github.com/clearlinux/clr-installer/log"
+	"github.com/clearlinux/clr-installer/model"
+	"github.com/clearlinux/clr-installer/progress"
+)
+
+// NetworkTestDialog is dialog window use to stop all other
+// user interaction while the network configuration is tested
+// for connectivity to the swupd server.
+// Check for:
+// - Working network interface
+// - Use of Proxy (if set)
+// - use of Swupd Mirror (if set)
+type NetworkTestDialog struct {
+	DialogBox *clui.Window
+	onClose   func()
+
+	modelSI       *model.SystemInstall
+	progressBar   *clui.ProgressBar
+	progressMax   int
+	resultLabel   *clui.Label
+	dismissButton *SimpleButton
+}
+
+// Success is part of the progress.Client implementation and represents the
+// successful progress completion of a task by setting
+// the progress bar to "full"
+func (dialog *NetworkTestDialog) Success() {
+	dialog.progressBar.SetValue(dialog.progressMax)
+	clui.RefreshScreen()
+}
+
+// Failure is part of the progress.Client implementation and represents the
+// unsuccessful progress completion of a task by setting
+// the progress bar to "fail"
+func (dialog *NetworkTestDialog) Failure() {
+	bg := dialog.progressBar.BackColor()
+	flashTime := 100 * time.Millisecond
+	//dialog.progressBar.SetValue(0)  // leave the bar where it fails?
+	for i := 1; i <= 5; i++ {
+		dialog.progressBar.SetBackColor(term.ColorRed)
+		clui.RefreshScreen()
+		time.Sleep(flashTime)
+		dialog.progressBar.SetBackColor(bg)
+		clui.RefreshScreen()
+		time.Sleep(flashTime)
```

Comment:
sleeping and refreshing twice?

Reply:
This is the "Flashing" between Red and the default color to make the failure obvious to the user. We have the same code in Failure code for the Progress bar during the installation code.